.gitlab/docker/Dockerfile.spack_compiler

```diff
@@ -7,25 +7,22 @@ SHELL ["/bin/bash", "-c"]
 # CSCS CI's .container-builder
 RUN spack config --scope site add  concretizer:targets:host_compatible:false
 
+# Create an anonymous spack environment
+RUN spack env create pika_env
+
```

Comment:
With the removal of `-e pika_env` from below, I'd move this into `Dockerfile.spack_dependencies`.